ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/linear_optimisation/algorithms/IteratingLinearOptimizer.java

```diff
@@ -94,13 +94,13 @@ public static LinearOptimizationResult optimize(IteratingLinearOptimizerInput in
                     iteration,
                     input.getObjectiveFunction()
             );
-            lastResult = currentResult;
+            previousResult = currentResult;
 
-            Map<Boolean, IteratingLinearOptimizationResultImpl> mipShouldStop = updateMipAndLogResult(parameters.getCapPstVariation(), linearProblem, input, iteration, currentResult, bestResult);
-            if (mipShouldStop.containsKey(Boolean.TRUE)) {
+            Pair<IteratingLinearOptimizationResultImpl, Boolean> mipShouldStop = updateBestResultAndCheckStopCondition(parameters.getDecreasePstRange(), linearProblem, input, iteration, currentResult, bestResult);
+            if (mipShouldStop.getValue().equals(Boolean.TRUE)) {
```

Comment:
The concept of key & value has no sense here. use left & right instead
Also you don't need the ".equals(Boolean.TRUE)"
```suggestion
            if (mipShouldStop.getRight()) {
```